client/src/main/java/com/linecorp/decaton/client/DecatonClientImpl.java

```diff
@@ -77,6 +77,19 @@ public DecatonClientImpl(String topic,
         return producer.sendRequest(key, request);
     }
 
+    @Override
+    public CompletableFuture<PutTaskResult> put(String key, T task, TaskMetaData overrideTaskMetaData) {
+        byte[] serializedTask = serializer.serialize(task);
+
+        DecatonTaskRequest request =
+                DecatonTaskRequest.newBuilder()
+                                  .setMetadata(convertToTaskMetadataProto(overrideTaskMetaData))
+                                  .setSerializedTask(ByteString.copyFrom(serializedTask))
+                                  .build();
+
+        return producer.sendRequest(key, request);
```

Comment:
let's add a private method like below and use it also from here https://github.com/line/decaton/pull/72/files#diff-b62f1c35402d8557a005fbb36c449954296e0efcb21d473196e04029d4c6fc33R65 to avoid duplicating common parts.

```java
private CompletableFuture<PutTaskResult> put(String key, T task, TaskMetadataProto metadataProto)
```